### ISSUE TITLE
Vopono forget password after failed connection attempt

### ISSUE THREAD

**Issue by Nyanraltotlapun** [jamesmcm/vopono#288]
Vopono forget password after failed connection attempt.
Currently I am using vopono with NordVPN under Arch Linux, and it often fails to connect to random server.
After failing, it forgets vpn service password and I need to enter it again.
It is a major pain for me.

Maybe I am missing something?

If I am not, I propose to change this behavior in some way:
1. Just don't forget password and login.
2. Or make option to alter this behaivour (forget or no)

Is something from this sounds reasonable? If it is, may I work on it and send pull request?

**Comment by jamesmcm** [comment]
Are you using OpenVPN? If you set up the server config with vopono it should save the OpenVPN password in `~/.config/vopono/nordvpn/openvpn/auth.txt`

If it's a custom config you can do the same thing with the OpenVPN config file - https://forums.openvpn.net/viewtopic.php?t=11342

**Comment by Nyanraltotlapun** [comment]
> Are you using OpenVPN? If you set up the server config with vopono it should save the OpenVPN password in `~/.config/vopono/nordvpn/openvpn/auth.txt`
> 
> If it's a custom config you can do the same thing with the OpenVPN config file - https://forums.openvpn.net/viewtopic.php?t=11342

I am creating config by `vopono sync`
Choosing NordVPN
And entering login and password when asked.

After this, I am using vopono as follows:
`vopono exec -p NordVPN -s county_name --no-proxy "firefox"`

When vopono fails to connect to server, next time when I issuing same command it start asking to enter login and password for NordVPN.

**Comment by Nyanraltotlapun** [comment]
> Are you using OpenVPN? If you set up the server config with vopono it should save the OpenVPN password in `~/.config/vopono/nordvpn/openvpn/auth.txt`
> 
> If it's a custom config you can do the same thing with the OpenVPN config file - https://forums.openvpn.net/viewtopic.php?t=11342

Deletion of auth file is happening here - https://github.com/jamesmcm/vopono/blob/62b6026be02a2fe9db421bfe2874376bd51fa1af/vopono_core/src/network/openvpn.rs#L161

It seems that its looking for "AUTH_FAILED" in OpanVPN log. Maybe its false positive?

**Comment by jamesmcm** [comment]
Yeah, that makes sense, the aim was to delete the auth file to stop it retrying if the password is wrong and re-prompt the user.

It'd be good to make the existing behaviour an option / separate command though.

**Comment by Nyanraltotlapun** [comment]
There is logs from failed connection:
```
1735419895.723899 40 DEPRECATED OPTION: --cipher set to 'AES-256-CBC' but missing in --data-ciphers (AES-256-GCM:AES-128-GCM:CHACHA20-POLY1305). OpenVPN ignores --cipher for cipher negotiations. 
1735419895.723919 3 Note: '--allow-compression' is not set to 'no', disabling data channel offload.
1735419895.723934 40 WARNING: file '/home/kira/.config/vopono/nordvpn/openvpn/auth.txt' is group or others accessible
1735419895.723940 1 OpenVPN 2.6.12 [git:makepkg/038a94bae57a446c+] x86_64-pc-linux-gnu [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [PKCS11] [MH/PKTINFO] [AEAD] [DCO] built on Dec 16 2024
1735419895.723948 1 library versions: OpenSSL 3.4.0 22 Oct 2024, LZO 2.10
1735419895.723961 1 DCO version: N/A
1735419895.724068 40 WARNING: --ping should normally be used with --ping-restart or --ping-exit
1735419895.724645 1 TCP/UDP: Preserving recently used remote address: [AF_INET]149.22.89.34:1194
1735419895.724670 2b000003 Socket Buffers: R=[212992->212992] S=[212992->212992]
1735419895.724675 1 UDPv4 link local: (not bound)
1735419895.724679 1 UDPv4 link remote: [AF_INET]149.22.89.34:1194
1735419895.759440 14000003 TLS: Initial packet from [AF_INET]149.22.89.34:1194, sid=df5c7719 92d85105
1735419895.759465 40 WARNING: this configuration may cache passwords in memory -- use the auth-nocache option to prevent this
1735419895.831175 14000002 VERIFY OK: depth=2, C=PA, O=NordVPN, CN=NordVPN Root CA
1735419895.831318 14000002 VERIFY OK: depth=1, O=NordVPN, CN=NordVPN CA9
1735419895.831441 14000002 VERIFY KU OK
1735419895.831448 14000002 Validating certificate extended key usage
1735419895.831453 14000002 ++ Certificate has EKU (str) TLS Web Server Authentication, expects TLS Web Server Authentication
1735419895.831457 14000002 VERIFY EKU OK
1735419895.831461 14000002 VERIFY X509NAME OK: CN=ch434.nordvpn.com
1735419895.831465 14000002 VERIFY OK: depth=0, CN=ch434.nordvpn.com
1735419895.865337 14000002 Control Channel: TLSv1.3, cipher TLSv1.3 TLS_AES_256_GCM_SHA384, peer certificate: 4096 bits RSA, signature: RSA-SHA512, peer temporary key: 253 bits X25519
1735419895.865361 1 [ch434.nordvpn.com] Peer Connection Initiated with [AF_INET]149.22.89.34:1194
1735419895.865371 14000003 TLS: move_session: dest=TM_ACTIVE src=TM_INITIAL reinit_src=1
1735419895.865401 14000003 TLS: tls_multi_process: initial untrusted session promoted to trusted
1735419897.091697 22000003 SENT CONTROL [ch434.nordvpn.com]: 'PUSH_REQUEST' (status=1)
1735419897.236995 0 AUTH: Received control message: AUTH_FAILED
1735419897.237146 1 SIGTERM[soft,auth-failure] received, process exiting
```

**Comment by jamesmcm** [comment]
Thanks, it's a pain there's no way to distinguish it from a real invalid credentials error.

But this change is made in PR #289 - hopefully I'll have time to release it this week.

**Comment by jamesmcm** [comment]
The auth file deletion issue is fixed in release v0.10.11

**Comment by Nyanraltotlapun** [comment]
> The auth file deletion issue is fixed in release v0.10.11

Thank you! Vopono is really important to me, I am appreciating you work very much!
